File: fileprovider-camera/camera/src/main/java/com/example/camera/library/FileProviderUtils.java
package com.example.camera.library;

import android.content.Context;
import android.net.Uri;
import android.support.v4.content.FileProvider;

import java.io.File;

public final class FileProviderUtils {

    private static final String AUTHORITIES_SUFFIX = ".provider";
    private static final String FILE_PROVIDER_PATHS = "temp";

    public static Uri getFileProviderUri(final Context context, final String fileName) {
        return FileProvider.getUriForFile(context,
                context.getPackageName() + AUTHORITIES_SUFFIX,
                createFile(context, fileName));
    }

    @SuppressWarnings("ResultOfMethodCallIgnored")
    public static void deleteFile(final Context context, final String fileName) {
        final File file = createFile(context, fileName);
        file.delete();
    }

    @SuppressWarnings("ResultOfMethodCallIgnored")
    private static File createFile(final Context context, final String fileName) {
        final File fileProviderDir = new File(context.getFilesDir(), FILE_PROVIDER_PATHS);
        if (!fileProviderDir.exists()) {
            fileProviderDir.mkdirs();
        }
        return new File(fileProviderDir, fileName);
    }
}

File: fileprovider-camera/camera/src/main/java/com/example/camera/library/FileProviderUtils.kt
package com.example.camera.library

import android.content.Context
import android.net.Uri
import androidx.core.content.FileProvider
import java.io.File

object FileProviderUtils {

    private const val AUTHORITIES_SUFFIX = ".provider"
    private const val FILE_PROVIDER_PATHS = "temp"

    fun getFileProviderUri(context: Context, fileName: String): Uri =
        FileProvider.getUriForFile(
            context,
            context.packageName + AUTHORITIES_SUFFIX,
            createFile(context, fileName),
        )

    fun deleteFile(context: Context, fileName: String) =
        createFile(context, fileName).delete()

    private fun createFile(context: Context, fileName: String): File {
        val fileProviderDir = File(context.filesDir, FILE_PROVIDER_PATHS)
        if (!fileProviderDir.exists()) {
            fileProviderDir.mkdirs()
        }
        return File(fileProviderDir, fileName)
    }
}
